### ISSUE TITLE
FunctionMetadataProviderGenerator code fails to build if project name has dash in name

### ISSUE THREAD

**Issue by mlhickey** [Azure/azure-functions-dotnet-worker#2076]
Given a project with name IM-CFS-AADLCM, FunctionMetadataProviderGenerator  creates a namespace of:
`namespace IM-CFS-AADLCM`
which fails during build:
![image](https://github.com/Azure/azure-functions-dotnet-worker/assets/20215068/e962a17d-e674-465d-8676-e8240cc9b1d5)

Changing the dash character to underscore, i.e. IM_CFS_AADLCM creates a namespace of:
`namespace IM_CFS_AADLCM`
which successfully builds.  Is this expected behavior?

**Comment by kshyju** [comment]
As per the [specs](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/namespaces#143-namespace-declarations), namespaces cannot have `-` in it. 

If not overridden, The `RootNamespace` msbuild property value is populated from the project name. You are seeing the issue because your `RootNamespace` has `-` in it and the auto generated types are defined in a namespace which is same as the `RootNamespace` value by default. You can override this behavior by explicitly specifying another valid namespace name in the `FunctionsGeneratedCodeNamespace` property. See #1930 for details.

To resolve your issue, either change your `RootNamespace`  to a value without a dash in it or explicitly specify a value for `FunctionsGeneratedCodeNamespace` msbuild property.

`<FunctionsGeneratedCodeNamespace>TestFunction</FunctionsGeneratedCodeNamespace>`

Let us know if you still run into issues.

**Comment by mlhickey** [comment]
I guess the challenge is that the namespace is being derived from the project name by FunctionMetadataProviderGenerator , not inherent within the code itself.  If would seem appropriate to either collapse the generated namespace or generate a unique value rather than generate a non-compliant name.

**Comment by kshyju** [comment]
@mlhickey Agree. We will look into making a fix to handle this and release and update to the package.

**Comment by kshyju** [comment]
@mlhickey 

Thank you for your patience. We released a new version of the worker.sdk package which includes a fix for this issue. Please upgrade `Microsoft.Azure.Functions.Worker.Sdk` package to [1.16.3](https://www.nuget.org/packages/Microsoft.Azure.Functions.Worker.Sdk/1.16.3) or later.

https://github.com/Azure/azure-functions-dotnet-worker/releases/tag/sdk-1.16.3

Should you encounter any further challenges with the dotnet isolated model, please do not hesitate to reach out. Thank you for your understanding and cooperation.
